backend/inspirehep/legacy/views.py

```diff
@@ -117,17 +118,17 @@ def redirect_info(info_path):
     return redirect(f"{current_app.config['LEGACY_BASE_URL']}/info/{info_path}", 302)
 
 
-@blueprint.route("/legacy/arxiv/<arxiv>", methods=("GET",))
+@blueprint.route("/legacy/arxiv/<path:arxiv>", methods=("GET",))
 def redirect_to_lit_record_by_arxiv(arxiv):
     recid = get_pid_for_pid("arxiv", arxiv, "recid")
     if not recid:
-        abort(404)
+        jsonify({"message": "Record not found"}), 404
```

Comment:
No, it shouldn't be a JSON response, it should redirect to the UI 404 page, as this is not an API route.